pkg/motor/update_bucket.go

```diff
@@ -8,26 +8,24 @@ import (
 
 	"github.com/sonr-io/sonr/internal/bucket"
 	"github.com/sonr-io/sonr/pkg/client"
-	"github.com/sonr-io/sonr/pkg/motor/types"
-	mt "github.com/sonr-io/sonr/pkg/motor/types"
 	"github.com/sonr-io/sonr/pkg/tx"
 	bt "github.com/sonr-io/sonr/x/bucket/types"
 )
 
-func (mtr *motorNodeImpl) UpdateBucket(request mt.UpdateBucketRequest) (bucket.Bucket, error) {
-	if request.Creator == "" {
-		return nil, errors.New("Invalid Address")
+func (mtr *motorNodeImpl) UpdateBucket(label string, did string, role bt.BucketRole, visibility bt.BucketVisibility, content []*bt.BucketItem) (bucket.Bucket, error) {
```

Comment:
Why are we pulling everything out of the request model only to put it back? This will require changes in binding layers too

Reply:
I can revert to using an object in the parameters for the binding layers